ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,493 @@
+                infile_ext = os.path.splitext(path)[1]
+                infile_ext = infile_ext.lower()
+                is_fits = False
+
+                if 'fit' in infile_ext:
+                    is_fits = True
+
+            # Only list FITS files unless it is Ginga generated buffer
+            if not is_fits:
+                continue
+
+            # Only list modified buffers
+            if mod_only and t is None:
+                continue
+
+            # More than one ext modified, append to existing entry
+            if infile in treedict:
+                if t is not None:
+                    treedict[infile].extlist.add(idx)
+                    elist = sorted(treedict[infile].extlist)
+                    treedict[infile].MODEXT = ';'.join(
+                        map(self._format_extname, elist))
+
+            # Add new entry
+            else:
+                if t is None:
+                    s = ''
+                    extlist = set()
+                else:
+                    s = self._format_extname(idx)
+                    extlist = set([idx])
+                treedict[infile] = Bunch.Bunch(
+                    IMAGE=infile, MODEXT=s, extlist=extlist, path=path)
+
+        self.treeview.set_tree(treedict)
+
+        # Resize column widths
+        n_rows = len(treedict)
+        if n_rows == 0:
+            self.w.status.set_text('Nothing available for saving')
+        elif n_rows < self.settings.get('max_rows_for_col_resize', 5000):
+            self.treeview.set_optimal_column_widths()
+            self.logger.debug('Resized columns for {0} row(s)'.format(n_rows))
+
+        return
+
+    def update_channels(self):
+        """Update the GUI to reflect channels and image listing.
+        """
+        if not self.gui_up:
+            return
+
+        self.logger.debug("channel configuration has changed--updating gui")
+        try:
+            channel = self.fv.get_channelInfo(self.chname)
+
+        except KeyError:
+            channel = self.fv.get_channelInfo()
+
+        if channel is None:
+            raise ValueError('No channel available')
+
+        self.chname = channel.name
+
+        w = self.w.channel_name
+        w.clear()
+
+        self.chnames = list(self.fv.get_channelNames())
+        #self.chnames.sort()
+        for chname in self.chnames:
+            w.append_text(chname)
+
+        # select the channel that is the current one
+        try:
+            i = self.chnames.index(channel.name)
+        except IndexError:
+            i = 0
+        self.w.channel_name.set_index(i)
+
+        # update the image listing
+        self.redo()
+
+    def select_channel_cb(self, w, idx):
+        self.chname = self.chnames[idx]
+        self.logger.debug("channel name changed to '%s'" % (self.chname))
+        self.redo()
+
+    def keys_from_datasrc(self):
+        """Yield back keys for image listing from Ginga's data cache."""
+        channel = self.fv.get_channelInfo()
```

Comment:
I'll let you fix this up if you want.  Personally, I think it complicates reading code to use `yield` in cases where a not very large iterable is built.  And it may slow down the code as well slightly.  But I'm not overly concerned about it in cases where the code is not in a hotspot.